packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -622,13 +622,20 @@ export default class CodeGen {
                 for (const attribute of node.attributes) {
                     const { name, value } = attribute;
                     if (isExpression(value)) {
-                        this.staticExpressionMap.set(attribute, `a${partId}:${name}`);
-                        attributeExpressions.push(
-                            t.property(
-                                t.literal(name),
-                                bindAttributeExpression(attribute, node, this, false)
-                            )
-                        );
+                        if (name === 'class') {
+                            this.staticExpressionMap.set(attribute, `c${partId}:`);
```

Comment:
I plan to update the usage of `WeakMap` here based on the [comments](https://github.com/salesforce/lwc/pull/4056#discussion_r1525331628) from #4056.

Reply:
We reverted back to use the `WeakMap` since the AST is exposed externally.